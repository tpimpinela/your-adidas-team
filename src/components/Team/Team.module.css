.team {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 12px;
  max-width: 150px;
  padding: 2rem 3rem;
  transition: 0.2s color, 0.2s background-color;
  cursor: pointer;

  &:nth-child(1n) {
    background-color: var(--background-color);
  }

  &:nth-child(2n) {
    background-color: var(--second-background-color);
  }

  &:hover {
    background-color: white;
    color: black;
  }

  & .team__logo {
    width: 100%;
    max-height: 100px;
    object-fit: fill;
  }
}
