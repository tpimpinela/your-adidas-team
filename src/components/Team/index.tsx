import { TeamMapped } from "../../models/teams.models";
import styles from "./Team.module.css";

interface Props {
  team: TeamMapped;
  onTeamClick: (teamId: number) => void;
}

const Team = ({ team, onTeamClick }: Props) => {
  return (
    <div
      data-testid="team"
      onClick={() => onTeamClick(team.id)}
      className={styles.team}
    >
      <img
        className={styles.team__logo}
        src={team.logo}
        alt={`Logo of ${team.name} team`}
      />
      <span>{team.name}</span>
    </div>
  );
};

export default Team;
