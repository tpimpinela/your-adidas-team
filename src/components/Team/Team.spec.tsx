import { render, RenderResult } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { beforeEach, describe, expect, Mock, vi, it } from "vitest";
import Team from ".";
import teamMappedMock from "../../mocks/teamMapped.mock";

describe("<Team />", () => {
  let renderResult: RenderResult;
  let onTeamClickMock: Mock<any>;

  beforeEach(() => {
    onTeamClickMock = vi.fn();
    renderResult = render(
      <Team team={teamMappedMock} onTeamClick={onTeamClickMock} />
    );
  });

  it("should match the snapshopt", () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it("should call onTeamClick when the team is clicked", async () => {
    await userEvent.click(renderResult.getByTestId("team"));
    expect(onTeamClickMock).toHaveBeenNthCalledWith(1, teamMappedMock.id);
  });
});
