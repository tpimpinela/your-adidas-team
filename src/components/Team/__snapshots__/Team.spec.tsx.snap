// Vitest Snapshot v1

exports[`<Team /> > should match the snapshopt 1`] = `
<div>
  <div
    class="_team_953c9b"
    data-testid="team"
  >
    <img
      alt="Logo of Test team mapped team"
      class="_team__logo_953c9b"
      src="http://test.com/logo.png"
    />
    <span>
      Test team mapped
    </span>
  </div>
</div>
`;
